pkg/service/service.go

```diff
@@ -121,9 +123,24 @@ func getResolver(ctx context.Context, configFile *configfile.ConfigFile) (remote
 	}
 
 	authOpts := []docker.AuthorizerOpt{docker.WithAuthCreds(credFunc)}
+
+	// Append insecure client if enabled
+	if true {
+		authOpts = append(authOpts, docker.WithAuthClient(newInsecureClient()))
+	}
+
 	authorizer := docker.NewDockerAuthorizer(authOpts...)
+
+	var h docker.RegistryHosts
+	// Append plainHttp Opt if enabled
+	if true {
+		h = docker.ConfigureDefaultRegistries(docker.WithAuthorizer(authorizer), docker.WithPlainHTTP(docker.MatchAllHosts))
```

Comment:
There is a difference between plain HTTP, i.e. no encryption and TLS without verification. We may encounter both scenarios. I think that typically self-signed certificates would fall into the second category of TLS without verification.

Reply:
Currently I tried to achive 100% feature parity with nerdctl.
In nerdctl, the `--insecure-registry` flag enables HTTP and disables TLS check.

Do you think we need to make this to 2 separate flag?

Reply:
Could you add that comment to the code, so that we have a source for the method? Link to the file and line with a GitHub permalink.